docs/proposals/20220411-cluster-api-state-metrics.md

```diff
@@ -0,0 +1,323 @@
+| `capi_kubeadmcontrolplane_owner`                                 | `1`                                            | Gauge | `owner_kind=<owner kind>` `owner_name=<owner name>` | [ReplicaSet] |
+
+#### MachineDeployment CR
+
+Common labels:
+
+- `machinedeployment=<machinedeployment-name>`
+
+| metric name                                                          | value                                         | type  | additional labels/tags                              | xref                              |
+|----------------------------------------------------------------------|-----------------------------------------------|-------|-----------------------------------------------------|-----------------------------------|
+| `capi_machinedeployment_labels`                                      | `1`                                           | Gauge | `.metadata.labels`                                  | common                            |
+| `capi_machinedeployment_created`                                     | `.metadata.creationTimestamp`                 | Gauge |                                                     | common                            |
+| `capi_machinedeployment_status_replicas`                             | `.status.replicas`                            | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_status_replicas_available`                   | `.status.availableReplicas`                   | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_status_replicas_unavailable`                 | `.status.unavailableReplicas`                 | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_status_replicas_updated`                     | `.status.updatedReplicas`                     | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_spec_replicas`                               | `.spec.replicas`                              | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_spec_strategy_rollingupdate_max_unavailable` | `.spec.strategy.rollingUpdate.maxUnavailable` | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_spec_strategy_rollingupdate_max_surge`       | `.spec.strategy.rollingUpdate.maxSurge`       | Gauge |                                                     | [Deployment]                      |
+| `capi_machinedeployment_status_phase`                                | `.status.phase==<phase>`                      | Gauge | `phase=<phase>`                                     | [Pod], [machinedeployment phases] |
```

Comment:
I think we'll want to track some 1 particular conditions here instead

Reply:
I also added the condition metric for the other resources. Looks like I also missed adding the initial metrics for MHC too..

Thanks for the review 👍